packages/crayons-core/src/utils/utils.ts

```diff
@@ -83,3 +83,16 @@ export const handleKeyDown =
       handler(event);
     }
   };
+
+export const throttle = (func, context, delay = 1000) => {
```

Comment:
One point to consider here is that the throttling in this implementation is at the mercy of how busy the main thread is. If it's very busy, the calls to this method may end up getting throttled with a longer duration between two successful calls (since the lock could get released a lot later than 1s).

One alternative idea I can think of is to store a timestamp instead of a lock. In subsequent calls, compare the present time to the last successful execution's timestamp. If it's above the `delay` threshold, proceed with execution and update the timestamp, if not, drop the execution.